### PR TITLE
NFSD Bike for NFSD Fun Category

### PR DESCRIPTION
**Pull request by BramvanZijp**
## About the PR
Swaps the Civillian hoverbike in the fun part of the loadout tab for the NFSD Variant if playing an NFSD role, costs 25k.

## Why / Balance
Its an NFSD hoverbike, I feel like if you buy a bike roundstart as NFSD you should get it. And the 10k discount is because theres enough expenses on NFSD already anyway.

## How to test
Run the build in a release config and check the loadouts tab of any NFSD role if you have atleast 50 hours of playtime, it should show the NFSD bike under the fun tab instead of the civ one.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- tweak: The Hoverbike under the fun tab of the loadouts menu has been replaced with the NFSD variant for NFSD personell, keeping the 50 hours playtime requirement.

### CODE REVIEW

File: Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml [hunk 1]
```diff
@@ -641,6 +641,30 @@
   - ContractorHoverbikeFlatpack
   - ContractorEmotionalPetCarrier
 
+- type: loadoutGroup
+  id: NFSDFun # Left Hand - 1 only, variant with the NFSD HoverBike.
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml:645]: Ok find me another place in the code where NFSD isnt Nfsd on entity

Also isnt its a fucking insane over kill to add a full louadout just for this?

File: Resources/Prototypes/_NF/Loadouts/Jobs/Contractor/fun.yml [hunk 2]
```diff
@@ -225,6 +225,19 @@
   inhand:
     - HoverbikeFlatpack
 
+- type: loadout
+  id: NFSDHoverbikeFlatpack
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Loadouts/Jobs/Contractor/fun.yml:229]: NFSD?
Nfsd

File: Resources/Prototypes/_NF/Loadouts/Jobs/Contractor/fun.yml [hunk 3]
```diff
@@ -225,6 +225,19 @@
...
+  equipment: NFSDHoverbikeFlatpack
+  effects:
+  - !type:GroupLoadoutEffect
+    proto: ContractorT3
+  price: 25000
+
+- type: startingGear
+  id: NFSDHoverbikeFlatpack
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Loadouts/Jobs/Contractor/fun.yml:237]: NFSD?
Nfsd

**Review by dvir001 — CHANGES_REQUESTED** [review]

**Comment by whatston3** [comment]
I suppose the in-context question is who owns the hoverbike?  The one you get from an uplink is NFSD property, sure.  Is this a personal bring-along?  Isn't the rest of your NFSD gear free?

File: Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml (hunk 1 above)
Comment (whatston3) [reply Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml:645]: Unfortunately:
`NFSDCriminalRecordsComputerCircuitboard`
`NFSDAdvancedRadarConsoleCircuitboard`

**Comment by dvir001** [comment]
Im not a fan of adding a full loadout setup just for a single item that exists on the uplink, this is more maintainer upkeep that we dont need.

You can just lower the price on the uplink if you see it as an issue.

File: Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml (hunk 1 above)
Comment (dvir001) [reply Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml:645]: ┻━┻ ︵ヽ(`Д´)ﾉ︵ ┻━┻
This need a fix in a new PR

Comment (dvir001) [reply Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml:645]: > > Ok find me another place in the code where NFSD isnt Nfsd on entity
> > Also isnt its a fucking insane over kill to add a full loadout just for this?
> 
> As for the full loadout, unless you want hoverbikes to get their own tab, I dont see an easy way to do this. I made sure to keep the name of the tab the same, so its only a backend thing.

Honestly just lower the price on uplink

Comment (BramvanZijp) [reply Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml:645]: > > > Ok find me another place in the code where NFSD isnt Nfsd on entity
> > > Also isnt its a fucking insane over kill to add a full loadout just for this?
> > 
> > 
> > As for the full loadout, unless you want hoverbikes to get their own tab, I dont see an easy way to do this. I made sure to keep the name of the tab the same, so its only a backend thing.
> 
> Honestly just lower the price on uplink

What would be a good price though? Like 2-3?

Comment (dvir001) [reply Resources/Prototypes/_NF/Loadouts/contractor_loadout_groups.yml:645]: > > > > Ok find me another place in the code where NFSD isnt Nfsd on entity
> > > > Also isnt its a fucking insane over kill to add a full loadout just for this?
> > > 
> > > 
> > > As for the full loadout, unless you want hoverbikes to get their own tab, I dont see an easy way to do this. I made sure to keep the name of the tab the same, so its only a backend thing.
> > 
> > 
> > Honestly just lower the price on uplink
> 
> What would be a good price though? Like 2-3?

Make it 3

**Comment by whatston3** [comment]
> Maybe theres a way to create a seperate loadout thats paranted by the fun tab and thus wouldnt require adding new items to both fun and nfsdfun?

If I understand you correctly, the issue is that you're modifying one of the data fields in the object - you can't say "inherit the parent's values and add these", you need to specify the entire set.

A separate question: what if the NFSD roles had _a bunch_ of NFSD-specific loadout items, maybe even between multiple loadout groups?  As it stands, you're proposing just the bike replacement, but what if there was a larger set of them?  e.g. NFSD balloon, handheld flag, toy sheriff's badge, maybe some commemorative NFSD pinpointer that points to the NFSD outpost.

**Comment by GiggIess** [comment]
I must weigh in here as a player, I'm not a fan of this. I like to be able to buy a bike that lets me place and take off my own bag. The NFSD bikes are "upgraded" and forced to carry satchels, that you cannot remove.

**Comment by dvir001** [comment]
Price lowered to 3 on new upstream update soon :tm:

**Pull request closed without merging**